### ISSUE TITLE
`clippy::suspicious_operation_groupings` creates code that fails to compile

### ISSUE THREAD

**Issue by liquidhelium** [rust-lang/rust-clippy#13137]
### Summary

I was using clippy when I found it generated incorrect code, so I made a minimal reproducer.

### Lint Name

clippy::suspicious_operation_groupings

### Reproducer

```rust
fn bug() {
    #[derive(Clone, Copy)]
    struct A {
        value: f32,
        time: f32,
    }
    struct T(f32);
    let a = A {
        value:1.0,
        time:2.0
    };
    let a1 = a;
    let b = T(2.0);

    let t = (b.0 - a.time) / (a1.time - a.time); // this line

}
```

### Version

```text
rustc 1.79.0 (129f3b996 2024-06-10)
binary: rustc
commit-hash: 129f3b9964af4d4a709d1383930ade12dfe7c081
commit-date: 2024-06-10
host: x86_64-unknown-linux-gnu
release: 1.79.0
LLVM version: 18.1.7
```

**Comment by alex-semenyuk** [comment]
It seems it's not an issue at 1.80 https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=68be92da92699514d85efa6162744743

**Comment by IceTDrinker** [comment]
hitting that as well

offending code

```rust
pub fn pbs_variance_132_bits_security_gaussian_impl(
    input_lwe_dimension: f64,
    output_glwe_dimension: f64,
    output_polynomial_size: f64,
    decomposition_base: f64,
    decomposition_level_count: f64,
    modulus: f64,
) -> f64 {
    input_lwe_dimension
        * (2.06537277069845e-33
            * decomposition_base.powf(2.0)
            * decomposition_level_count
            * output_polynomial_size.powf(2.0)
            * (output_glwe_dimension + 1.0)
            + decomposition_level_count
                * output_polynomial_size
                * ((4.0 - 2.88539008177793 * modulus.ln()).exp2()
                    + (-0.0497829131652661 * output_glwe_dimension * output_polynomial_size
                        + 5.31469187675068)
                        .exp2())
                * ((1_f64 / 12.0) * decomposition_base.powf(2.0) + 0.166666666666667)
                * (output_glwe_dimension + 1.0)
            + (1_f64 / 12.0) * modulus.powf(-2.0)
            + (1_f64 / 2.0)
                * output_glwe_dimension
                * output_polynomial_size
                * (0.0208333333333333 * modulus.powf(-2.0)
                    + 0.0416666666666667
                        * decomposition_base.powf(-2.0 * decomposition_level_count))
            + (1_f64 / 24.0) * decomposition_base.powf(-2.0 * decomposition_level_count))
}
```

also the allow() does not work on nightly-2024-10-03


rustc -Vv
```
rustc 1.83.0-nightly (18b1161ec 2024-10-02)
binary: rustc
commit-hash: 18b1161ec9eeab8927f91405bca0ddf59a4a26c9
commit-date: 2024-10-02
host: x86_64-unknown-linux-gnu
release: 1.83.0-nightly
LLVM version: 19.1.0
```
